src/lib/holocene/modal.svelte

```diff
@@ -21,45 +26,82 @@
     dispatch('cancelModal');
   };
 
-  const handleKeyUp = (event: KeyboardEvent) => {
+  const handleKeyboardNavigation = (event: KeyboardEvent) => {
+    if (!open) {
+      return;
+    }
+
     if (event.key === 'Escape') {
       cancelModal();
+      return;
+    }
+
+    if (event.key === 'Tab') {
+      if (event.shiftKey) {
+        if (document.activeElement === closeButton) {
+          confirmButton.focus();
+          event.preventDefault();
+        }
+      } else if (document.activeElement === confirmButton.buttonElement) {
+        closeButton.focus();
+        event.preventDefault();
+      }
     }
   };
+
+  $: {
+    if (open && modalElement) {
+      modalElement.focus();
+    }
+  }
 </script>
 
-<svelte:window on:keyup={handleKeyUp} />
+<svelte:window on:keydown={handleKeyboardNavigation} />
 {#if open}
-  <div class="modal">
+  <div bind:this={modalElement} class="modal">
     <div on:click={cancelModal} class="overlay" />
-    <div class="body" class:large>
+    <div
```

Comment:
We actually want the `bind:this={modalElement}` on this `div` as it has the modal contents and doesn't include the overlay.